vnext/ReactUWP/Base/UwpReactInstance.h

```diff
@@ -58,13 +58,17 @@ class UwpReactInstance : public IReactInstance, public ::std::enable_shared_from
   void SetXamlViewCreatedTestHook(std::function<void(react::uwp::XamlView)> testHook) override;
   void CallXamlViewCreatedTestHook(react::uwp::XamlView view) override;
 
+  void SetReactControl(std::weak_ptr<ReactControl> reactControl) override;
+  virtual std::weak_ptr<ReactControl> GetReactControl() override;
+
   // Public functions
   std::shared_ptr<facebook::react::MessageQueueThread> GetNewUIMessageQueue() const;
 
 private:
   void OnHitError(const std::string& error) noexcept;
 
 private:
+  std::weak_ptr<ReactControl> m_reactControl;
```

Comment:
This ties a UwpReactInstance to a single ReactControl, but this can be a 1 to many relationship.

Reply:
Randy, Can you make an example of this case? I think it always attach to one control, then detach if we re-use it.
Is there any way to get the reactControl from UIManager?

Reply:
I create m_tagsToXamlReactControl in NativeUIManager, for any node, first look up its parent from shadow node tree, then get the react control.

---
In reply to: [299194676](https://github.com/microsoft/react-native-windows/pull/2706#discussion_r299194676) [](ancestors = 299194676)